src/aerie_cli/commands/command_context.py

```diff
@@ -1,17 +1,37 @@
 import typer
 from aerie_cli.aerie_client import AerieClient
 from aerie_cli.utils.sessions import get_active_session_client
+from aerie_cli.aerie_host import AerieHostConfiguration
 
 app = typer.Typer()
 
 class CommandContext:
-    hasura_admin_secret = None
+    hasura_admin_secret: str = None
+    configuration: AerieHostConfiguration = None
+    __client: AerieClient = None
+
     def __init__(self) -> None:
         raise NotImplementedError
 
     @classmethod
     def get_client(cls) -> AerieClient:
-        client = get_active_session_client()
+        """Get the AerieClient for any command's execution.
+        
+        Returns:
+            AerieClient
+        """
+        # If the configuration was set in the CLI by the user,
+        # then the returned client will be derived from that configuration.
+        if cls.configuration != None and cls.__client == None:
+            session = cls.configuration.start_session()
+            cls.__client = AerieClient(session)
+
+        if cls.__client == None:
+            # no configuration specified in CLI, so the active session will be used instead
+            cls.__client = get_active_session_client()
+            cls.configuration = cls.__client.host_session.configuration_name
+
         if cls.hasura_admin_secret:
-            client.host_session.session.headers["x-hasura-admin-secret"] = cls.hasura_admin_secret
-        return client
\ No newline at end of file
+            cls.__client.host_session.session.headers["x-hasura-admin-secret"] = cls.hasura_admin_secret
+
+        return cls.__client
```

Comment:
Is there any reason why we store `__client`? I think we can just have a `client` variable that's local to this method and gets returned.

Reply:
If we don't store the client somewhere, the client will be instantiated every time `get_client` is called. I don't think this will ever change the expected behavior of the function aside from that, but I could change it to a local variable if it makes the code more legible/organized.

Reply:
I think we only ever call `get_client` once per command though, right?

Reply:
That is true, adding another variable to CommandContext might not be the best idea then. It isn't a necessary optimization in that case.

I've made the change in 0d1d167